client/landing/login/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Global polyfills
+ */
+import 'boot/polyfills';
+import { hydrate, render } from 'controller/web-util';
+
+/**
+ * External dependencies
+ */
+import page from 'page';
+/**
+ * Internal dependencies
+ */
+import createStore from './store';
+import { setupMiddlewares } from './common';
+import initLoginSection from 'login';
+
+// goofy import for environment badge, which is SSR'd
+import 'components/environment-badge/style.scss';
+
+// Create Redux store
+const store = createStore();
+
+setupMiddlewares( store );
+
+page( '*', ( context, next ) => {
+	context.store = store;
```

Comment:
The `context.store` is already done in the `setupMiddlewares` function, both on enter and exit. Creating small focused page.js handlers is a good direction, so I'd recomment removing the code from `setupMiddlewares`.

Reply:
Done!